docs/events.md

```diff
@@ -68,3 +78,129 @@ point in time:
+from a specific date or from a previous stream position.  To start from the earliest available events (~1 year),
+pass `streamPosition = '0'`.  The stream will fetch all past events as quickly as your listener consumes them.
+Once the stream catches up to the current time, it will begin polling for new events every `pollingInterval` seconds
+(default = 60).
+
+```js
+client.events.getEnterpriseEventStream({
+    startDate: '2016-01-01T00:00:00-08:00',
+    pollingInterval: 60
+}, callback);
+```
+
+Note that Box buffers enterprise events for ~60 seconds before making them available to the `/events` API
+(to ensure that events are delivered in-order and without duplicates), so polling with an interval of less than
+60 seconds is not normally needed.
+
+If you pass `pollingInterval = 0`, then the stream will not use polling, but will end when all of the currently
+available events have been delivered.
+
+```js
+client.events.getEnterpriseEventStream({
+        startDate: '2016-01-01T00:00:00-08:00',
+        endDate: '2017-01-01T00:00:00-08:00',
+        pollingInterval: 0
+    }, function(err, stream) {
+	
+    if (err) { // Handle error }
+
+    stream.on('end', function(event) {
+        // Reached the end of the stream.
+    });
+});
+```
+
+You can also filter the event stream to only receive specific event types.  The set of enterprise event types
+is available in `client.events.enterpriseEventTypes`.
+
+```js
+const stream = client.events.getEnterpriseEventStream({
+    eventTypeFilter: [client.events.enterpriseEventTypes.UPLOAD, client.events.enterpriseEventTypes.DOWNLOAD]
+}, callback);
+```
+
+Since `EnterpriseEventStream` implements [stream.Readable](https://nodejs.org/api/stream.html#stream_readable_streams),
+you can use the usual flow-control mechanisms on the stream:
+
+```js
+stream.pause();
+stream.resume();
+stream.isPaused();
+```
+
+You can also pipe the output to a [stream.Writable](https://nodejs.org/api/stream.html#stream_writable_streams) stream
+(it must be an "object mode" stream):
+
+```js
+stream.pipe(writableObjectModeStream);
+```
+
+Handling errors
+---------------
+If an API or network error occurs, the stream will ignore the error and continue polling at the usual rate until
+the connection can be re-established.  You can respond to errors with an `'error'` event listener:
+
+```js
+stream.on('error', function(event) {
+    // Handle the error.
+});
+```
+
+Persisting the Stream State
+---------------------------
+
+In many applications, you may need to persist the stream state so that you can resume processing events from the
+same point if your application is interrupted and restarted.  You can attach a `'streamPosition'` event listener
+to be notified each time the stream position changes.
+
+```js
+client.events.getEnterpriseEventStream(function(err, stream) {
+	
+    if (err) { // Handle error }
+
+    // Restore the stream state from the previous run.
+    stream.setStreamState(readState());
```

Comment:
It seems a little odd to get a stream and then set its state manually; perhaps we should add an `initialState` option to `getEnterpriseEventStream()` so the user can just pass in the state and have it set automatically.